### PR TITLE
Swipes

### PR DESCRIPTION
**Pull request by chrisdugne**
### inter question

Screen Shot 2019-04-10 at 2 25 40 PM

### authentication

Screen Shot 2019-04-10 at 4 48 30 PM

### Testing Strategy

- [x] Manual testing
- [x] Unit testing

### CODE REVIEW

**Comment by chrisdugne** [comment]
[ETQ Coorp, sur chaque élément swipable, je remonte un hit](https://trello.com/c/jQkYw1tn/1253-etq-coorp-sur-chaque-%C3%A9l%C3%A9ment-swipable-je-remonte-un-hit)

File: src/components/cards.js [hunk 1]
```diff
@@ -10,10 +10,18 @@ export type Card = {|
   title: string,
   type: CardType,
   resource?: Resource,
-  offeringExtraLife?: boolean
+  offeringExtraLife?: boolean,
+  analyticsData?: {
```
Comment (VincentCATILLON) [inline src/components/cards.js]: can you use `analyticsParams` to use the same terms everywhere ?

File: src/components/cards.js [hunk 2]
```diff
@@ -10,10 +10,18 @@ export type Card = {|
...
+    element: string,
```
Comment (VincentCATILLON) [inline src/components/cards.js]: what is element ?

File: src/components/cards.js [hunk 3]
```diff
@@ -10,10 +10,18 @@ export type Card = {|
...
+    isCorrect: boolean,
+    offeringExtraLife?: boolean,
```
Comment (VincentCATILLON) [inline src/components/cards.js]: `offeringExtraLife ` is already defined in props. redundant

File: src/components/cards.js [hunk 4]
```diff
@@ -10,10 +10,18 @@ export type Card = {|
...
+    type: string,
```
Comment (VincentCATILLON) [inline src/components/cards.js]: what is type ?

File: src/components/correction.js [hunk 5]
```diff
@@ -123,15 +123,44 @@ class Correction extends React.PureComponent<Props> {
     } = this.props;
     const correctionCard: Card = {
       type: CARD_TYPE.CORRECTION,
-      title: translations.correction
+      title: translations.correction,
+      analyticsData: {
+        element: 'card-interquestion',
```
Comment (VincentCATILLON) [inline src/components/correction.js]: I used deck-card

File: src/components/correction.js [hunk 6]
```diff
@@ -123,15 +123,44 @@ class Correction extends React.PureComponent<Props> {
...
+        isCorrect,
+        type: 'correction'
+      }
+    };
+    const tipCard = {
+      type: CARD_TYPE.TIP,
+      title: translations.didYouKnowThat,
+      analyticsData: {
+        element: 'card-interquestion',
```
Comment (VincentCATILLON) [inline src/components/correction.js]: I used deck card

File: src/components/correction.js [hunk 7]
```diff
@@ -123,15 +123,44 @@ class Correction extends React.PureComponent<Props> {
...
+        isCorrect,
+        type: 'tip-card'
+      }
+    };
+
+    const keyPointCard = {
+      type: CARD_TYPE.KEY_POINT,
+      title: translations.keyPoint,
+      analyticsData: {
+        element: 'card-interquestion',
```
Comment (VincentCATILLON) [inline src/components/correction.js]: I used deck-card

File: src/components/correction.js [hunk 8]
```diff
@@ -123,15 +123,44 @@ class Correction extends React.PureComponent<Props> {
...
+        isCorrect,
+        type: 'key-point'
```
Comment (VincentCATILLON) [inline src/components/correction.js]: Don't you think that type is already defined in props (the `type` prop is the same thing)

File: src/containers/cards-scalable.js [hunk 9]
```diff
@@ -5,8 +5,13 @@ import DeckSwiper from '@coorpacademy/react-native-deck-swiper';
 
 import Cards from '../components/cards';
 import type {Card as CardType} from '../components/cards';
+import {ANALYTICS_EVENT_TYPE} from '../const';
+
+import withAnalytics from './with-analytics';
+import type {WithAnalyticsProps} from './with-analytics';
 
 type Props = {|
```
Comment (VincentCATILLON) [inline src/containers/cards-scalable.js]: Can you please add `$Exact<>`

File: src/containers/carousel.js [hunk 10]
```diff
@@ -49,6 +52,7 @@ const styles: GenericStyleProp = StyleSheet.create({
 });
 
 type Props = {|
```
Comment (VincentCATILLON) [inline src/containers/carousel.js]: Can you please add `$Exact<>`

File: src/containers/carousel.js [hunk 11]
```diff
@@ -98,6 +102,16 @@ class Carousel extends React.PureComponent<Props, State> {
   };
 
   handleItemChange = (index: number) => {
+    const {analytics} = this.props;
+
+    analytics &&
+      analytics.logEvent(ANALYTICS_EVENT_TYPE.SWIPE, {
+        element: 'card-authentication',
+        step: `from ${this.state.items[this.state.currentItemIndex].iconName} to ${
```
Comment (VincentCATILLON) [inline src/containers/carousel.js]: Not asked but don't know if it's relevant and exploitable

File: src/types.js [hunk 12]
```diff
@@ -90,5 +91,5 @@ export type AnalyticsEventType =
 
 export type AnalyticsUserProperty = 'id' | 'brand';
 export type AnalyticsEventParams = {
-  [key: string]: string | number
+  [key: string]: string | number | boolean
```
Comment (VincentCATILLON) [inline src/types.js]: boolean is not supported, only string and numbers are. you have to cast booleans into numbers

File: src/containers/carousel.js (hunk 11 above)
Comment (chrisdugne) [reply src/containers/carousel.js]: without this info you cant tell what step you've swiped and the way it was swiped
with `iconName` it's readable and usable:

File: src/components/cards.js (hunk 1 above)
Comment (chrisdugne) [reply src/components/cards.js]: removed to use props instead

File: src/containers/carousel.js (hunk 11 above)
Comment (VincentCATILLON) [reply src/containers/carousel.js]: IMO we should split it in few properties to be exploitable (previous_index, current_index)

Comment (chrisdugne) [reply src/containers/carousel.js]:
```
        id: 'authentication-card',
        from: `${this.state.items[this.state.currentItemIndex].iconName}`,
        to: `${this.state.items[index].iconName}`
```

**Comment by VincentCATILLON** [comment]
Good job dude

**Pull request merged**